tests/AutoReview/ProjectCodeTest.php

```diff
@@ -348,6 +348,61 @@ public function testThatTestClassCoversAreCorrect(string $testClassName): void
         }
     }
 
+    /**
+     * @dataProvider provideTestClassCases
+     */
+    public function testThereIsNoPhpVersionUsedDirectly(string $className): void
+    {
+        // should only shrink, baseline of violations on moment of adding this test
+        $exceptions = [
+            \PhpCsFixer\Tests\Fixer\CastNotation\LowercaseCastFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\CastNotation\ShortScalarCastFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\ClassNotation\ClassDefinitionFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\FunctionNotation\NativeFunctionInvocationFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\FunctionNotation\RegularCallableCallFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\LanguageConstruct\NoUnsetOnPropertyFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\Operator\AssignNullCoalescingToCoalesceEqualFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\Operator\IncrementStyleFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\Operator\OperatorLinebreakFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\Whitespace\NoExtraBlankLinesFixerTest::class,
+            \PhpCsFixer\Tests\Fixer\Whitespace\NoSpacesAroundOffsetFixerTest::class,
+        ];
+
+        $classesAllowedToUsePhpVersionIdConstant = [
+            \PhpCsFixer\Tests\FixerDefinition\VersionSpecificationTest::class,
+            \PhpCsFixer\Tests\FixerDefinition\VersionSpecificCodeSampleTest::class,
+        ];
+        if (\in_array($className, $exceptions, true)) {
+            $this->expectNotToPerformAssertions();
+
+            return;
+        }
+
+        $tokens = $this->createTokensForClass($className);
+
+        $stringTokens = array_filter(
+            $tokens->toArray(),
+            static fn (Token $token): bool => $token->isGivenKind(T_STRING)
+        );
+
+        $strings = array_map(
+            static fn (Token $token): string => $token->getContent(),
+            $stringTokens
+        );
+
+        $strings = array_unique($strings);
+        $message = sprintf('%s must not used directly.', $className);
+        self::assertNotContains('phpversion', $strings, $message);
```

Comment:
i'm not convinced to prevent usage in _whole_ test file
ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7575#issuecomment-1858894164